### ISSUE TITLE
Pre-Publish Checklist: Do not suggest adding captions for videos without audio tracks

### ISSUE THREAD

**Issue by BrittanyIRL** [GoogleForCreators/web-stories-wp#7371]
Found during bug bash .7 by @barklund 
This could use a little bit of discussion before we decide what to do with this.

>  I’ve added a background video without sound. The checklist recommends that I add captions. That’s very annoying! We should remove this check for videos without sound. And maybe even allow users to remove this check for videos with no relevant sound (e.g. if the only sound is waves crashing, it’s not important to capture in captions, I guess). Maybe a checkbox in the captions panel that says “This video has no (important) audio”?

In future version(s) of the checklist it's been discussed to have a 'toggle' to silence recommendations that aren't helpful. That might be a potential solution to this.

**Comment by swissspidy** [comment]
IIRC this also came up in the support forums before. Definitely would be good to address.

**Comment by swissspidy** [comment]
Silencing/Ignoring/Hiding recommendations would also be useful for #5627

For captions, @spacedmonkey mentioned that we could also try to make this information (whether a video has no sound) available via the REST API, so that muted videos could be automatically marked as such and we wouldn't even show a PPC warning.

In a simpler form, users could mark a video as muted in the Design panel, which could have the same effect.

**Comment by spacedmonkey** [comment]
Looking the REST API for media, here are two examples. 

### Video with audio
<img width="267" alt="Screenshot 2021-05-05 at 17 11 51" src="https://user-images.githubusercontent.com/237508/117173797-093bab00-adc5-11eb-9d41-ca226b612e4d.png">

### Video without audio
<img width="231" alt="Screenshot 2021-05-05 at 17 12 01" src="https://user-images.githubusercontent.com/237508/117173805-0b9e0500-adc5-11eb-8878-31fba4aaaefa.png">

Sadly this data is not in the media picker admin ajax request, it we could add this in our plugin. 

In [getResourceFromAttachment](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/app/media/utils/getResourceFromAttachment.js) / [getResourceFromMediaPicker](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/app/media/utils/getResourceFromMediaPicker.js) we could check for the `media_details.audio`. If the field is present, then set `hasAudio` true / false.  If `hasAudio` is set to false, hide checks for PPC captions and even the caption upload dialog. 

This would only work for newer video element, as the data would not be populated in old video elements. So maybe default to true for older video elements.

**Comment by swissspidy** [comment]
@spacedmonkey How feasible would it be to add this data to the media picker admin ajax request? We already filter `wp_prepare_attachment_for_js`, so could we just extend that?

**Comment by swissspidy** [comment]
^ @spacedmonkey friendly ping

**Comment by BrittanyIRL** [comment]
_circling back_ 

I'd love to get this sorted out! 

@spacedmonkey thoughts on @swissspidy 's ask about feasibility to add data to media picker admin ajax request? 

otherwise, your idea to check for `media_details.audio` in the `getResource...` (respectfully), makes sense and would require no extra work from the infra pod (besides review) as we would just update those files.

**Comment by spacedmonkey** [comment]
@swissspidy Worth noting that as of https://github.com/google/web-stories-wp/pull/7691 the exact same data in `media_details` are in both the REST API and admin ajax. Example from admin ajax. 

![Screenshot 2021-06-08 at 17 07 50](https://user-images.githubusercontent.com/237508/121219945-19d7b900-c87c-11eb-979a-dd329a9e6320.png)

In my testing, all transcoded video, had an audio object in the apis. Sadly we can't easily migrate existing data, as it would require a REST API. 

I would recommend adding a field of `hasAudio`. This would default to true on existing videos and read audio object. From other media types like image and gif, it would default to false.

**Comment by swissspidy** [comment]
> In my testing, all transcoded video, had an audio object in the apis. Sadly we can't easily migrate existing data, as it would require a REST API.

The only issue would be when this data is missing, but the video actually _**does**_ have audio.
That would lead to a case where we mute the video (because we think it has no sound) and the user has no idea why it's muted.

In other words, how reliably available is this data?

**Comment by spacedmonkey** [comment]
>In other words, how reliably available is this data

I need to so some more testing, but from what I can see the transcoded videos all seem to pick up audio when processed by WordPress. I would need to check with different video files and file types.

**Comment by spacedmonkey** [comment]
Another question, do we hide the caption option all together for soundless videos.

**Comment by swissspidy** [comment]
> Another question, do we hide the caption option all together for soundless videos.

As per #7730: no

**Comment by swissspidy** [comment]
This is mostly covered by #7885, #7665, and #7730. Closing in favor of these three dedicated tickets.
